Formula/volta.rb

```diff
@@ -0,0 +1,27 @@
+class Volta < Formula
+  desc "Volta: The JavaScript Launcher"
+  homepage "https://volta.sh"
+  url "https://github.com/volta-cli/volta.git", :revision => "10a9bee148d2a833069d6cc27aea56af3092dc6e"
+  version "0.7.2"
+
+  depends_on "rust" => :build
+
+  conflicts_with "hadoop", :because => "both install `yarn` binaries"
+  conflicts_with "node", :because => "both install `node` binaries"
+  conflicts_with "npm", :because => "both install `npm` binaries"
+  conflicts_with "yarn", :because => "both install `yarn` binaries"
+
+  def install
+    ENV["VOLTA_HOME"] = prefix.to_s
+    system "./dev/unix/volta-install.sh", "--release", "--skip-setup"
+  end
+
+  def postinstall
+    chmod "+x", "#{opt_prefix}/bin/volta"
+  end
+
+  test do
+    system "#{bin}/volta", "install", "node@12.16.1"
```

Comment:
Does this ship it's own node binaries?

Reply:
It doesn't ship it, it actually compiles from source when you ask it to install one of the versions (like `nvm` does, for example)

Reply:
Any chance of a test that takes builds something a bit smaller?

Reply:
I could try to use `volta completion` to generate the completion file and check if the file was generated, if that's enough. If it's best to actually build something to ensure Formula quality, then node is the best (and I think only) option.

Reply:
How long does a build typically take? If >5m: I think the completion file and checking something in the contents (or that Bash/ZSH can parse it) would be good.

Reply:
It takes a few seconds, but it does require the user to be connected to the internet.

```
time volta install node@12.16.2
success: installed and set node@12.16.2 (with npm@6.14.4) as default
volta install node@12.16.2  1.02s user 2.25s system 77% cpu 4.233 total
```

Reply:
Oh, so it's downloading binaries? That's fine, then. Will it still work on Linux, out of interest? (not a blocker if not)

Reply:
Yeah, the installer is the same for macOS and Linux. It should probably work for Linux Brew as is, if that's why you're asking.